src/resources/api-resource.js

```diff
@@ -1,6 +1,22 @@
 import rp from 'request-promise-native';
 
 export default class ApiResource {
+  /**
+   * Clean parameters object by filtering out undefineds.
+   *
+   * @param {Object} params the parameters object.
+   * @returns {Object} the clean object of parameters
+   * @static
+   * @memberof ApiResource
+   */
+  static cleanParameters = params =>
+    Object.keys(params).reduce((returns, element) => {
+      if (params[element] !== undefined) {
+        return { ...returns, [element]: params[element] };
+      }
+      return returns;
+    }, {});
+
```

Comment:
@kevinadda What do you think about using the `cleanParameters` function like this:

```javascript
this.rp = options =>
      rp({		     
        ...baseOptions,
        ...options,
        qs: cleanParameters(options.qs)
      });
```

This way you don't have to call `cleanParameters` each time (this prevents you from sending undefined params, but I think this is good?)